draft-barnes-mls-protocol.md

```diff
@@ -1435,44 +1434,128 @@ all arrive at the following state:
 
 # Message Protection
 
-[[ OPEN ISSUE: This section has initial considerations about message
-protection.  This issue clearly needs more specific recommendations,
-possibly a protocol specification in this document or a separate
-one. ]]
-
-The primary purpose of this protocol is to enable an authenticated
-group key exchange among participants.  In order to protect messages sent among
-those participants, an application will need to specify how messages are protected.
-
-For every epoch, the root key of the ratcheting tree can be used to
-derive key material for symmetric operations such as encryption/AEAD and MAC;
-AEAD or MAC MUST be used to ensure that the message originated from a member
-of the group.
-
-In addition, asymmetric signatures SHOULD be used to authenticate the sender
-of a message.
+The primary purpose of the handshake protocol is to enable an authenticated
+group key exchange among participants.  In order to protect Application messages sent
+among those participants, an application will need to specify how messages
+are protected. The message protection layer is somewhat independant of the chosen
+group key-establishment mechanism as long as this mechanism provides the
+expected cryptographic security properties for the Message Protection Layer.
+
+All Application messages SHOULD be protected with an Authenticated-Encryption
+with Associated-Data (AEAD) encryption scheme using AEAD keys derived from the
+Application secret (ie. see the key schedule figure).
+Note that "Authenticated" in this context is not mean messages are known to
+be sent by a specific participant but only from a legitimate member of the group.
+To obtain non-repudiability, Handshake messages MUST use asymmetric signatures
+to strongly authenticate the sender of a message, Application messages SHOULD
+use such signatures to provide the same property.
+
+For each change to the Group root secret of the ratcheting tree, participants
+MUST derive new Application secret.
+The main reason for that is to make sure that only the intended participants,
+according to the latest group update, will be able to derive the authenticated
+encryptions keys from the Application secret.
+
+Application secrets can be independantly refreshed on a regular basis. This is
+for two main reasons: the first is to improve FS and PCS guarantees for the messages if
+an Application AEAD key gets compromised; the second is to avoid key exhaustion
+from AEAD encrypting more data under the same keys than expected by the encryption scheme.
+The policy to update this secret is currently left to the application but it
+is RECOMMENDED for this secret to be updated with every message.
+Since Application AEAD keys are automatically updated at each Group operation,
+it is not expected for the Clients to be very agressive on independant updates.
+In all cases, participant MUST refresh their Application secret before
+exhaustion of the associated AEAD keys.
```

Comment:
This para is more for security considerations.  It doesn't really seem appropriate to discuss PCS in this context.  You get PCS by replacing the app secret when you update the group secret; nothing you do within an epoch affects PCS.

Reply:
Also, let's just go ahead and define a concrete schedule for updating these things, as discussed below.